src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.jsx

```diff
@@ -308,39 +326,85 @@ const DataSeriesFormItem = ({
-          <MultiSelect
-            // need to re-gen if selected card changes or if a dataItem is removed from the list
-            key={`data-item-select-${removedDataItems.length}-selected_card-id-${cardConfig.id}`}
-            id={`${cardConfig.id}_dataSourceIds`}
-            label={mergedI18n.selectDataItems}
-            direction="bottom"
-            itemToString={(item) => item.id}
-            initialSelectedItems={initialSelectedItems}
-            items={formatDataItemsForDropdown(validDataItems)}
-            light
-            onChange={({ selectedItems }) => {
-              // need to remove the category if the card is a stacked timeseries bar
-              const card =
-                cardConfig.content.type === BAR_CHART_TYPES.STACKED &&
-                cardConfig.content.timeDataSourceId &&
-                selectedItems.length > 1
-                  ? omit(cardConfig, 'content.categoryDataSourceId')
-                  : cardConfig;
+        isComplexDataSeries ? (
+          <div className={`${baseClassName}--input`}>
+            <MultiSelect
+              // need to re-gen if selected card changes or if a dataItem is removed from the list
+              key={`data-item-select-${removedDataItems.length}-selected_card-id-${cardConfig.id}`}
+              id={`${cardConfig.id}_dataSourceIds-select`}
+              label={mergedI18n.selectDataItems}
+              direction="bottom"
+              itemToString={(item) => item.id}
+              initialSelectedItems={initialSelectedItems}
+              items={formatDataItemsForDropdown(validDataItems)}
+              light
+              onChange={({ selectedItems }) => {
+                const newCard = handleDataSeriesChange(
+                  selectedItems,
+                  cardConfig,
+                  setEditDataSeries
+                );
+                setSelectedDataItems(selectedItems.map(({ id }) => id));
+                onChange(newCard);
+              }}
+              titleText={mergedI18n.dataItem}
+              translateWithId={translateWithId}
+            />
+          </div>
+        ) : (
+          <div className={`${baseClassName}--input`}>
+            <ComboBox
+              // need to re-gen if selected card changes or if a dataItem is removed from the list
+              key={`data-item-select-${removedDataItems.length}-selected_card-id-${cardConfig.id}`}
+              id={`${cardConfig.id}_dataSourceIds-combobox`}
+              items={formatDataItemsForDropdown(validDataItems)}
+              itemToString={(item) => item.id}
+              titleText={mergedI18n.dataItemEditorDataItemTitle}
+              addToList={false}
+              placeholder={mergedI18n.filter}
+              // clears out the input field after each selection
+              selectedItem={{ id: '', text: '' }}
+              onChange={(selectedItem) => {
+                // ignore the value added by the "enter" keypress
```

Comment:
What does this mean? Does enter not work then?

Reply:
Enter does work, but without this check, a duplicate item gets generated in the list. May be something to do with the underlying implementation of that ComboBox feature paired with how we're using it.